### ISSUE TITLE
Error: Uncaught (in promise) Http: connection closed before message completed

### ISSUE THREAD

**Issue by JumpLink** [alosaur/alosaur#168]
This seems to be a problem directly from Deno, see https://github.com/denoland/deno/issues/11595

But maybe a workaround can be implemented here until this bug is fixed? Or is there a way to catch these errors yourself?

*Edit*
Deno version: 1.15.2
Alosaur version: 0.34.0

The problem occurs when I render a template and quickly press the reload button in the browser several times.

**Comment by JumpLink** [comment]
@irustm Seems to be fixed with deno v1.15.2

**Comment by JumpLink** [comment]
Hi @irustm, I still have the problem. Here is a small example to reproduce the problem:

app.ts
```ts
import {
    App,
    Area,
    Controller,
    Get,
  } from "https://deno.land/x/alosaur@v0.35.0/mod.ts";

  const sleep = (ms: number) => {
    return new Promise(resolve => setTimeout(resolve, ms))
  }


  @Controller()
  export class ViewController {
    @Get()
    async text() {
      // Placeholder for some calculations or requests from other services
      // This makes the difference
      await sleep(200); 
      return "Hello World!"
    }
  }
  
  @Area({
    controllers: [ViewController],
  })
  export class ViewArea {}

  const app = new App({
    areas: [ViewArea],
  });

  console.info("Deno version: ", Deno.version);
  
  app.listen();
```

Run this with `deno run --allow-net app.ts`

Here is a video which shows the issue:

https://user-images.githubusercontent.com/1073989/145722868-984aa4d5-593a-4c86-a071-259adca4a96e.mp4

**Comment by irustm** [comment]
It seems to be a problem on the Deno side.

I can only insert a handling of this exception, but I understand that this will not solve the problem.

```ts
function sleep(ms: number) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

async function handle(conn: Deno.Conn) {
    const httpConn = Deno.serveHttp(conn);
    for await (const { respondWith } of httpConn) {
        await sleep(200);

        respondWith(new Response());
    }
}

const server = Deno.listen({ port: 8000 });

for await (const conn of server) {
    handle(conn);
}

```

**Comment by JumpLink** [comment]
@irustm Thank you for investigation. If you can catch the error so that the app does not crash, that would already be a big help 🙏

**Comment by JumpLink** [comment]
@irustm This issue seems to be fixed in the new HTTP server API: https://github.com/denoland/deno_std/pull/1128

Example
```ts
import { serveListener, Handler } from "https://deno.land/std/http/server.ts";

const sleep = (ms: number) => {
  return new Promise(resolve => setTimeout(resolve, ms))
}

const requestHandler: Handler = async (/*request, connInfo*/) => {
  await sleep(200);
  const resp = new Response("hello world");
  return resp;
}

const listener = Deno.listen({ port: 8000 });
serveListener(listener, requestHandler);
```

I can not reproduce the error here, can you switch to this new API?

The old API is available here: https://deno.land/std/http/server_legacy.ts

**Comment by irustm** [comment]
yes, that works well. it would take a lot of work to use std. 


I think I can do it this week or else.

**Comment by JumpLink** [comment]
Thank you very much! And thanks for working on Alosaur in general.  I really like your framework.  👍

**Comment by irustm** [comment]
I added exception handling to respondWith. Released it in v0.35.2 . I will wait for the Deno team to respond.

**Comment by irustm** [comment]
Deno issue:
https://github.com/denoland/deno/issues/13203

**Comment by irustm** [comment]
I inspected this behavior. it hides inside the `serve try catch` and closes the connection.

https://deno.land/std@0.120.0/http/server.ts#L308-L311

I will do the same and not display this error message

**Comment by irustm** [comment]
Published in v0.32.2 . Feel free to ask if you have any questions.
